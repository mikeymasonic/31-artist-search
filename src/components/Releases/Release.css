@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@200;300&display=swap');

.release {
  height: 20vw;
  width: 30vw;
  display: flex;
  justify-content: center;
  text-align: center;
  margin: 0 3vw 3vw 0;
  padding-top: 3vw;
  border: .2vw solid rgb(133, 220, 226);
  border-radius: 1vw;
  font-size: 1.4vw;
  font-family: 'Montserrat', sans-serif;
}

.release img {
  width: 10vw;
}

.link {
  text-decoration: none;
  color: #f7d89e;
  font-weight: 200;
}

.link:hover {
  color: white;
}

.release:hover {
  background: hotpink;
}

.titleh2 {
  font-size: 2vw;
  font-family: 'Montserrat', sans-serif;
  text-align: center;
  color: #f7d89e;
  font-weight: 200;
}

.titleh3 {
  font-size: 1vw;
  font-family: 'Montserrat', sans-serif;
  text-align: center;
  color: #f7d89e;
  font-weight: 200;
  margin-top: -1vw;
}