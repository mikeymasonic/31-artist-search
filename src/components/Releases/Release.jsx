import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Release.css';

const Release = ({
  releaseCover,
  releaseId,
  releaseTitle,
  releaseDate = 'unknown',
  artistName
}) => {
  
  return (
    <section className={styles.release}>
      <Link className={styles.link} to={`/songs/${artistName}/${releaseId}/${releaseTitle}`}>
        <img src={releaseCover} />
        <h2 className={styles.titleh2}>{releaseTitle}</h2> <h3 className={styles.titleh3}>released: {releaseDate}</h3>
      </Link>
    </section>
  );
};

Release.propTypes = {
  releaseId: PropTypes.string.isRequired,
  releaseTitle: PropTypes.string.isRequired,
  releaseDate: PropTypes.string.isRequired,
  releaseCover: PropTypes.string.isRequired,
  artistName: PropTypes.string.isRequired,
};

export default Release;
