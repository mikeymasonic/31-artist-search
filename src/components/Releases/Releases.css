.releases {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  padding: 0;
}

.releases li {
  list-style: none;
}