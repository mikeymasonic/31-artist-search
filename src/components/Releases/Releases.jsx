import React from 'react';
import PropTypes from 'prop-types';
import Release from './Release';
import placeholderImage from '../../assets/AlbumCoverPlaceholder.png';

const Releases = ({ releaseArray, artistName }) => {
  const releaseList = releaseArray.map(({ releaseId, releaseDate, releaseTitle, coverArtCount }) => (
    <li key={releaseId}>
      <Release
        releaseId={releaseId}
        releaseCover={coverArtCount ? `http://coverartarchive.org/release/${releaseId}/front-250` : placeholderImage }
        releaseDate={releaseDate}
        releaseTitle={releaseTitle}
        artistName={artistName}
      />
    </li>
  ));
  return (
    <ul>
      {releaseList}
    </ul>
  );
};

Releases.propTypes = {
  releaseArray: PropTypes.arrayOf(PropTypes.shape({
    releaseId: PropTypes.string.isRequired,
    releaseTitle: PropTypes.string.isRequired,
    releaseDate: PropTypes.string.isRequired,
    coverArtCount: PropTypes.bool.isRequired
  })).isRequired,
  artistName: PropTypes.string.isRequired
};

export default Releases;
