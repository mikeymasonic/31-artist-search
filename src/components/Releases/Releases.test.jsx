import React from 'react';
import { shallow } from 'enzyme';
import Releases from './Releases';

describe('Releases component', () => {
  it('renders Releases', () => {
    const wrapper = shallow(<Releases artistName='Silkworm' releaseArray={[{
      coverArtCount: true, releaseId: '311', releaseTitle: 'Firewater', releaseDate: 'May 4th, 1990' }, { coverArtCount: true, releaseId:'312', releaseTitle: 'Lifestyle', releaseDate: 'August 2nd, 1986' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
