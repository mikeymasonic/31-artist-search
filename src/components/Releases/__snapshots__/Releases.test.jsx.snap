// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Releases component renders Releases 1`] = `
<ul
  className="releases"
>
  <li
    key="311"
  >
    <Release
      artistName="Silkworm"
      releaseCover="http://coverartarchive.org/release/311/front-250"
      releaseDate="May 4th, 1990"
      releaseId="311"
      releaseTitle="Firewater"
    />
  </li>
  <li
    key="312"
  >
    <Release
      artistName="Silkworm"
      releaseCover="http://coverartarchive.org/release/312/front-250"
      releaseDate="August 2nd, 1986"
      releaseId="312"
      releaseTitle="Lifestyle"
    />
  </li>
</ul>
`;
